src/presentation/http/policies/userCanReadFileData.ts

```diff
@@ -0,0 +1,47 @@
+import hasProperty from '@infrastructure/utils/hasProperty.js';
+import type { PolicyContext } from '../types/PolicyContext.js';
+import { isEmpty, notEmpty } from '@infrastructure/utils/empty.js';
+import type { NoteInternalId } from '@domain/entities/note.js';
+
+/**
+ * Policy to check does user have permission to access file data,
+ * if file is a part of note
+ *
+ * @param context - Context, object containing Fatify request, Fastify reply and domain services
+ */
+export default async function userCanReadFileData(context: PolicyContext): Promise<void> {
```

Comment:
seems like this code works only for files in notes. So I'd suggest to name this policy more clear: like `userCanAccessNoteFile`

Reply:
no, it checks if the file belongs to the note by passed key, and if it does, it checks for access rights. It works for all files

Reply:
that statement should be determined by `type`, not the `noteId` existence

Reply:
Now i check type too

Reply:
I don't see working with type and location here.